architecture.md

```diff
@@ -0,0 +1,199 @@
+# Architecture
+
+- [Architecture](#architecture)
+  - [General](#general)
+  - [Activities and Fragments](#activities-and-fragments)
+  - [Data structure](#data-structure)
+    - [Repositories](#repositories)
+  - [Common features](#common-features)
+    - [RecyclerView](#recyclerview)
+    - [HTML-Content](#html-content)
+    - [Share](#share)
+    - [Refresh](#refresh)
+    - [Feedback to the user](#feedback-to-the-user)
+      - [Success/Error](#successerror)
+      - [Progress dialog](#progress-dialog)
+    - [Request permission](#request-permission)
+
+## General
+
+**IDE:** [Android Studio]  
+**Build system:** Gradle  
+**Programming language:** Kotlin  
+**Primary libraries:** [Jetpack], [Realm]
+
+- Use [LiveData] for a reactive data flow.
+- Use [KTX synthetic properties][ktx-synthetic] for accessing views from code.
+- Use [DataBinding] for displaying simple data. If a more complex expression is required for preparing/formatting the data, set it from code or use a static (`@JvmStatic`) utility method in the respective Fragment/Activity/ViewModel/View.
+- Use [ViewModel] for passing data from repository to view. If an Id is required (e.g. show data for a specific course), retrieve it using [`.viewmodels.IdViewModelFactory`][IdViewModelFactory].
+- Use [`.*Utils`][Utils]-classes, extension functions/properties and `suspend` when applicable!
+
+
```

Comment:
Just give a general architecture, package and tools overview here. Code examples are always taken from existing code. Also less to maintain here, separate documentation never works in such projects.